src/main/java/org/mifos/pheeBillPay/camel/routes/BillInquiryRouteBuilder.java

```diff
@@ -20,16 +22,28 @@ public class BillInquiryRouteBuilder extends ErrorHandlerRouteBuilder {
     @Autowired
     private Bill billDetails;
 
+    @Autowired
+    ResponseDTO responseDTO;
+
     @Override
     public void configure() {
 
         from("direct:bill-inquiry-response").routeId("bill-inquiry-response").log("Triggering callback for bill inquiry response")
                 .setHeader(Exchange.HTTP_RESPONSE_CODE, constant(200)).process(exchange -> {
-                    Object obj = exchange.getProperty(BILL_INQUIRY_RESPONSE);
+                    if(exchange.getProperty(BILL_INQUIRY_RESPONSE)!= null) {
+                        Object obj = exchange.getProperty(BILL_INQUIRY_RESPONSE);
                     ObjectMapper objectMapper = new ObjectMapper();
                     String jsonString = objectMapper.writeValueAsString(obj);
                     exchange.getIn().setBody(jsonString);
-                    logger.info("Bill Inquiry Responsejsostring: " + jsonString);
+                    logger.info("Bill Inquiry Response: " + jsonString);}
+                    else{
+                        exchange.getIn().setHeader(Exchange.HTTP_RESPONSE_CODE, constant(400));
+                        responseDTO = setFailureMessage(exchange);
+                        ObjectMapper objectMapper = new ObjectMapper();
+                        String jsonString = objectMapper.writeValueAsString(responseDTO);
+                        exchange.getIn().setBody(jsonString);
+                        logger.info("Bill Inquiry Failure Response: " + jsonString);
```

Comment:
use debug level logs,
also dont use + for concatenation use {} instead. This practice will make the code null safe